src/Source.cpp

```diff
@@ -10,16 +11,26 @@ SourcePtr Source::create(List spec) {
   std::string subclass(as<CharacterVector>(spec.attr("class"))[0]);
 
   int skip = as<int>(spec["skip"]);
-  std::string comment = as<std::string>(spec["comment"]);
+  SEXP comment_sexp = spec["comment"];
```

Comment:
Feel free to postpone this discussion if you want, but if we solve this now then I will do it right in the PR.

I don't see the benefit of using a `CharacterVector` here, as I need to set the encoding to UTF-8 anyway. This is a simple example (for a single element) using SEXP and CharacterVector. I will use the CharacterVector anyway if you prefer.

```
library(Rcpp)
cppFunction('std::string test1(SEXP x) {
  SEXP x_elm = STRING_ELT(x, 0);
  std::string x_elm_conv = Rf_translateCharUTF8(x_elm);
  Rcpp::Rcout << x_elm_conv << std::endl;
  return x_elm_conv;
}')

cppFunction('std::string test2(CharacterVector x) {
  Rcpp::String x_elm(x(0));
  x_elm.set_encoding(CE_UTF8);
  Rcout << x_elm.get_cstring() << std::endl;
  return x_elm.get_cstring();
}')


x <- "potato\u00f3"
test1(x)
test2(x)


x2 <- "potato\xf3"
Encoding(x2) <- "latin1"
test1(x2)
test2(x2)
```